### PR TITLE
losipiuk	Fix serializing nested types with decimal to JSON

### PR DESCRIPTION
**Pull request by losipiuk**
Fixes: https://github.com/prestodb/presto/issues/7328

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java [hunk 1]
```diff
@@ -197,4 +204,16 @@ public void testCastFromVarchar()
...
+    public void testCastFromNestedDecimalTypes() {
+        assertFunction("cast(ARRAY[DECIMAL '1234567890.1234567890'] as JSON)", JSON, "[1234567890.1234567890]");
+        assertFunction("cast(ARRAY[ARRAY[DECIMAL '1234567890.1234567890']] as JSON)", JSON, "[[1234567890.1234567890]]");
+        assertFunction("cast(MAP(ARRAY['abc'], ARRAY[DECIMAL '1234567890.1234567890']) as JSON)", JSON,
+                "{\"abc\":1234567890.1234567890}");
+        assertFunction("cast(MAP(ARRAY['abc'], ARRAY[MAP(ARRAY['def'], ARRAY[DECIMAL '1234567890.1234567890'])]) as JSON)", JSON,
+                "{\"abc\":{\"def\":1234567890.1234567890}}");
+        assertFunction("cast(ROW(DECIMAL '1234567890.1234567890') AS JSON)", JSON, "[1234567890.1234567890]");
+        assertFunction("cast(ROW(ROW(DECIMAL '1234567890.1234567890')) AS JSON)", JSON, "[[1234567890.1234567890]]");
+    }
```
Comment (maciejgrzybek) [inline presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java]: Add test for map where `array[decimal]` is key.

Comment (losipiuk) [inline presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java]: Can it be casted to JSON?

Comment (maciejgrzybek) [inline presto-main/src/test/java/com/facebook/presto/type/TestJsonOperators.java]: Ouch, fair. We don't have comparable arrays implemented in serializer.
Please disregard my comment than.

**Comment by wenleix** [comment]
This has the same root cause with https://github.com/prestodb/presto/issues/7226. Rather than fixing one specific type for a time, it might be better for us to redesign the `ArrayToJsonCast` to have the same behavior of the cast methods in `JsonFunctions`.

Cc @haozhun

**Comment by losipiuk** [comment]
@wenleix  I agree general solution would be better. I still think it is worthy to merge it as is. It does not add any complexity. Improves behaviour. In the meantime we can work on target solution if time allows.

**Comment by losipiuk** [comment]
@haozhun can we merge this short term fix

**Comment by cawallin** [comment]
@losipiuk -- I think this was subsumed by #7565; if so please close this PR

**Pull request closed without merging**